### PR TITLE
forwarded export features

### LINKED ISSUE #1624: Extract export forward information
### Summary

Feature that enables extraction of DLL export forwarding information (DLL Proxying)

Maybe it is possible to distinguish between if it is a relative path or absolute path in the export forward name when creating a CAPA rule.

Example rules:
export-forward-absolute-path.yml
export-forward-relative-path.yml
export-forward-system32.yml

Example of malicious use:
legit.exe -> that search for version.dll
version.dll -> Crafted version.dll with export forwards to original version.dll
vresion.dll -> Original version.dll

Example md5:
76fa734236daa023444dec26863401dc

### Motivation

To be able to make CAPA rule that will detect if DLL is leveraging export forwarding aka DLL proxying.

### PR DESCRIPTION
**Pull request by williballenthin**
closes #1624 

implementation currently supports:
  - [x] viv
  - [x] pefile
  - [x] ida
  - [ ] binja #1646 

### Checklist

- [ ] No CHANGELOG update needed

- [ ] No documentation update needed

### CODE REVIEW

**Comment by williballenthin** [comment]
IDA tests passing:

image

**Comment by williballenthin** [comment]
maybe we can defer the Binary Ninja implementation til @xusheng6 has a chance to look.

if this PR can be reviewed and merged by tomorrow then it should make the 6.0 release.

**Comment by xusheng6** [comment]
> maybe we can defer the Binary Ninja implementation til @xusheng6 has a chance to look.
> 
> if this PR can be reviewed and merged by tomorrow then it should make the 6.0 release.

I will try to deal with this next week, along with the unit test failure.

File: capa/features/extractors/viv/file.py [hunk 1]
```diff
@@ -25,10 +26,42 @@ def extract_file_embedded_pe(buf, **kwargs) -> Iterator[Tuple[Feature, Address]]
...
+            except UnicodeDecodeError:
+                continue
+
+            # use rpartition so we can split on separator between dll and name.
+            # the dll name can be a full path, like in the case of
+            # ef64d6d7c34250af8e21a10feb931c9b
+            # which i assume means the path can have embedded periods.
+            # so we don't want the first period, we want the last.
+            forwarded_dll, _, forwarded_symbol = forwarded_name.rpartition(".")
+            forwarded_dll = forwarded_dll.lower()
```
Comment (mr-tz) [inline capa/features/extractors/viv/file.py:59]: turn this into a util function, since it's repeated a few times?

**Review by mr-tz — APPROVED** [review]
one comment, but not a stopper

**Pull request merged**